File: flutter/ios/Classes/BunbuPresets.swift
import Foundation

struct BunbuPreset {
    let label: String
    let icon: String
    let code: String
}

let bunbuPresets: [BunbuPreset] = [
    BunbuPreset(
        label: "Todo List",
        icon: "checklist",
        code: """
import 'package:flutter/material.dart';

class Main extends StatefulWidget {
  Main();

  @override
  State<Main> createState() => _MainState();
}

class _MainState extends State<Main> {
  bool _d0 = false;
  bool _d1 = false;
  bool _d2 = false;
  bool _d3 = false;
  bool _d4 = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('My Tasks'),
        backgroundColor: Color(0xFF6C63FF),
      ),
      body: ListView(
        children: [
          GestureDetector(
            onTap: () { setState(() { _d0 = _d0 == false; }); },
            child: ListTile(
              leading: Icon(_d0 ? Icons.check_circle : Icons.radio_button_unchecked, color: _d0 ? Color(0xFF6C63FF) : Colors.grey),
              title: Text('Buy groceries', style: TextStyle(color: _d0 ? Colors.grey : Colors.black)),
            ),
          ),
          GestureDetector(
            onTap: () { setState(() { _d1 = _d1 == false; }); },
            child: ListTile(
              leading: Icon(_d1 ? Icons.check_circle : Icons.radio_button_unchecked, color: _d1 ? Color(0xFF6C63FF) : Colors.grey),
              title: Text('Walk the dog', style: TextStyle(color: _d1 ? Colors.grey : Colors.black)),
            ),
          ),
          GestureDetector(
            onTap: () { setState(() { _d2 = _d2 == false; }); },
            child: ListTile(
              leading: Icon(_d2 ? Icons.check_circle : Icons.radio_button_unchecked, color: _d2 ? Color(0xFF6C63FF) : Colors.grey),
              title: Text('Read a book', style: TextStyle(color: _d2 ? Colors.grey : Colors.black)),
            ),
          ),
          GestureDetector(
            onTap: () { setState(() { _d3 = _d3 == false; }); },
            child: ListTile(
              leading: Icon(_d3 ? Icons.check_circle : Icons.radio_button_unchecked, color: _d3 ? Color(0xFF6C63FF) : Colors.grey),
              title: Text('Write Flutter code', style: TextStyle(color: _d3 ? Colors.grey : Colors.black)),
            ),
          ),
          GestureDetector(
            onTap: () { setState(() { _d4 = _d4 == false; }); },
            child: ListTile(
              leading: Icon(_d4 ? Icons.check_circle : Icons.radio_button_unchecked, color: _d4 ? Color(0xFF6C63FF) : Colors.grey),
              title: Text('Go to the gym', style: TextStyle(color: _d4 ? Colors.grey : Colors.black)),
            ),
          ),
        ],
      ),
    );
  }
}
"""
    ),
    BunbuPreset(
        label: "Profile",
        icon: "person.circle",
        code: """
import 'package:flutter/material.dart';

class Main extends StatelessWidget {
  Main();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Profile'),
        backgroundColor: Color(0xFF2196F3),
      ),
      body: ListView(
        children: [
            SizedBox(height: 32),
            Center(
              child: Container(
                width: 100,
                height: 100,
                decoration: BoxDecoration(
                  color: Color(0xFF2196F3),
                  borderRadius: BorderRadius.all(Radius.circular(50)),
                ),
                child: Center(
                  child: Text('JD', style: TextStyle(fontSize: 36, color: Colors.white)),
                ),
              ),
            ),
            SizedBox(height: 16),
            Center(child: Text('Jane Doe', style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold))),
            SizedBox(height: 4),
            Center(child: Text('Flutter Developer', style: TextStyle(fontSize: 16, color: Colors.grey))),
            SizedBox(height: 24),
            Padding(
              padding: EdgeInsets.symmetric(horizontal: 32),
              child: Row(
                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                children: [
                  Column(
                    children: [
                      Text('142', style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold)),
                      Text('Posts', style: TextStyle(color: Colors.grey)),
                    ],
                  ),
                  Column(
                    children: [
                      Text('1.2k', style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold)),
                      Text('Followers', style: TextStyle(color: Colors.grey)),
                    ],
                  ),
                  Column(
                    children: [
                      Text('89', style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold)),
                      Text('Following', style: TextStyle(color: Colors.grey)),
                    ],
                  ),
                ],
              ),
            ),
            SizedBox(height: 24),
            Container(height: 1, color: Colors.grey),
            SizedBox(height: 8),
            ListTile(
              leading: Icon(Icons.email, color: Color(0xFF2196F3)),
              title: Text('jane.doe@email.com'),
            ),
            ListTile(
              leading: Icon(Icons.location_on, color: Color(0xFF2196F3)),
              title: Text('San Francisco, CA'),
            ),
            ListTile(
              leading: Icon(Icons.web, color: Color(0xFF2196F3)),
              title: Text('github.com/janedoe'),
            ),
          ],
        ),
    );
  }
}
"""
    ),
    BunbuPreset(
        label: "Weather",
        icon: "cloud.sun",
        code: """
import 'package:flutter/material.dart';

class Main extends StatelessWidget {
  Main();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Weather'),
        backgroundColor: Color(0xFF1565C0),
      ),
      body: Container(
        color: Color(0xFF1565C0),
        child: Column(
          children: [
            SizedBox(height: 40),
            Icon(Icons.wb_sunny, size: 80, color: Colors.amber),
            SizedBox(height: 16),
            Text('San Francisco', style: TextStyle(fontSize: 28, color: Colors.white)),
            SizedBox(height: 8),
            Text('72 F', style: TextStyle(fontSize: 64, fontWeight: FontWeight.normal, color: Colors.white)),
            Text('Sunny', style: TextStyle(fontSize: 18, color: Colors.grey)),
            SizedBox(height: 40),
            Padding(
              padding: EdgeInsets.symmetric(horizontal: 24),
              child: Row(
                mainAxisAlignment: MainAxisAlignment.spaceAround,
                children: [
                  Column(
                    children: [
                      Icon(Icons.water_drop, color: Colors.white),
                      SizedBox(height: 4),
                      Text('45%', style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold)),
                      Text('Humidity', style: TextStyle(color: Colors.grey, fontSize: 12)),
                    ],
                  ),
                  Column(
                    children: [
                      Icon(Icons.air, color: Colors.white),
                      SizedBox(height: 4),
                      Text('12 mph', style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold)),
                      Text('Wind', style: TextStyle(color: Colors.grey, fontSize: 12)),
                    ],
                  ),
                  Column(
                    children: [
                      Icon(Icons.visibility, color: Colors.white),
                      SizedBox(height: 4),
                      Text('10 mi', style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold)),
                      Text('Visibility', style: TextStyle(color: Colors.grey, fontSize: 12)),
                    ],
                  ),
                ],
              ),
            ),
            SizedBox(height: 40),
            Padding(
              padding: EdgeInsets.symmetric(horizontal: 16),
              child: Row(
                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                children: [
                  Column(
                    children: [
                      Text('Mon', style: TextStyle(color: Colors.grey)),
                      SizedBox(height: 4),
                      Icon(Icons.wb_sunny, color: Colors.amber, size: 28),
                      SizedBox(height: 4),
                      Text('74', style: TextStyle(color: Colors.white)),
                    ],
                  ),
                  Column(
                    children: [
                      Text('Tue', style: TextStyle(color: Colors.grey)),
                      SizedBox(height: 4),
                      Icon(Icons.cloud, color: Colors.grey, size: 28),
                      SizedBox(height: 4),
                      Text('68', style: TextStyle(color: Colors.white)),
                    ],
                  ),
                  Column(
                    children: [
                      Text('Wed', style: TextStyle(color: Colors.grey)),
                      SizedBox(height: 4),
                      Icon(Icons.cloud, color: Colors.grey, size: 28),
                      SizedBox(height: 4),
                      Text('65', style: TextStyle(color: Colors.white)),
                    ],
                  ),
                  Column(
                    children: [
                      Text('Thu', style: TextStyle(color: Colors.grey)),
                      SizedBox(height: 4),
                      Icon(Icons.wb_sunny, color: Colors.amber, size: 28),
                      SizedBox(height: 4),
                      Text('71', style: TextStyle(color: Colors.white)),
                    ],
                  ),
                  Column(
                    children: [
                      Text('Fri', style: TextStyle(color: Colors.grey)),
                      SizedBox(height: 4),
                      Icon(Icons.wb_sunny, color: Colors.amber, size: 28),
                      SizedBox(height: 4),
                      Text('75', style: TextStyle(color: Colors.white)),
                    ],
                  ),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }
}
"""
    ),
]

File: react-native/android/src/main/kotlin/dev/bunbu/bunbu/BunbuPresets.kt
package dev.bunbu.bunbu

data class BunbuPreset(
    val label: String,
    val code: String
)

val bunbuPresets = listOf(
    BunbuPreset(
        label = "Todo List",
        code = """
import React, { useState } from 'react';
import { View, Text, TouchableOpacity, FlatList, StyleSheet } from 'react-native';

const tasks = ['Buy groceries', 'Walk the dog', 'Read a book', 'Write React Native code', 'Go to the gym'];

export default function Main() {
  const [done, setDone] = useState(new Set());
  const toggle = (i) => {
    const next = new Set(done);
    next.has(i) ? next.delete(i) : next.add(i);
    setDone(next);
  };
  return (
    <View style={s.root}>
      <View style={s.header}><Text style={s.title}>My Tasks</Text></View>
      <FlatList
        data={tasks}
        keyExtractor={(_, i) => String(i)}
        renderItem={({ item, index }) => (
          <TouchableOpacity style={s.row} onPress={() => toggle(index)}>
            <Text style={s.icon}>{done.has(index) ? '✅' : '⬜'}</Text>
            <Text style={[s.label, done.has(index) && s.struck]}>{item}</Text>
          </TouchableOpacity>
        )}
      />
    </View>
  );
}

const s = StyleSheet.create({
  root: { flex: 1, backgroundColor: '#1A1A2E' },
  header: { padding: 16, backgroundColor: '#6C63FF' },
  title: { fontSize: 20, fontWeight: '700', color: '#fff' },
  row: { flexDirection: 'row', alignItems: 'center', padding: 16, borderBottomWidth: 1, borderBottomColor: '#2A2A3E' },
  icon: { fontSize: 20, marginRight: 12 },
  label: { fontSize: 16, color: '#E0E0E0' },
  struck: { textDecorationLine: 'line-through', color: '#666666' },
});
""".trimIndent()
    ),
    BunbuPreset(
        label = "Profile",
        code = """
import React from 'react';
import { View, Text, ScrollView, StyleSheet } from 'react-native';

export default function Main() {
  return (
    <ScrollView style={s.root}>
      <View style={s.header}>
        <View style={s.avatar}><Text style={s.initials}>JD</Text></View>
        <Text style={s.name}>Jane Doe</Text>
        <Text style={s.sub}>React Native Developer</Text>
      </View>
      <View style={s.stats}>
        {[['142','Posts'],['1.2k','Followers'],['89','Following']].map(([n,l])=>(
          <View key={l} style={s.stat}><Text style={s.statN}>{n}</Text><Text style={s.statL}>{l}</Text></View>
        ))}
      </View>
      <View style={s.info}>
        <Text style={s.infoRow}>📧  jane.doe@email.com</Text>
        <Text style={s.infoRow}>📍  San Francisco, CA</Text>
        <Text style={s.infoRow}>🔗  github.com/janedoe</Text>
      </View>
    </ScrollView>
  );
}

const s = StyleSheet.create({
  root: { flex: 1, backgroundColor: '#1A1A2E' },
  header: { alignItems: 'center', paddingVertical: 32, backgroundColor: '#2196F3' },
  avatar: { width: 100, height: 100, borderRadius: 50, backgroundColor: '#1976D2', justifyContent: 'center', alignItems: 'center' },
  initials: { fontSize: 36, color: '#fff', fontWeight: '700' },
  name: { fontSize: 24, fontWeight: '700', color: '#fff', marginTop: 12 },
  sub: { fontSize: 16, color: '#B3E5FC', marginTop: 4 },
  stats: { flexDirection: 'row', justifyContent: 'space-evenly', paddingVertical: 24 },
  stat: { alignItems: 'center' },
  statN: { fontSize: 20, fontWeight: '700', color: '#E0E0E0' },
  statL: { fontSize: 14, color: '#888888' },
  info: { padding: 24 },
  infoRow: { fontSize: 16, paddingVertical: 8, color: '#E0E0E0' },
});
""".trimIndent()
    ),
    BunbuPreset(
        label = "Weather",
        code = """
import React from 'react';
import { View, Text, ScrollView, StyleSheet } from 'react-native';

const days = [
  { day: 'Mon', icon: '☀️', temp: 74 },
  { day: 'Tue', icon: '☁️', temp: 68 },
  { day: 'Wed', icon: '☁️', temp: 65 },
  { day: 'Thu', icon: '☀️', temp: 71 },
  { day: 'Fri', icon: '☀️', temp: 75 },
];

export default function Main() {
  return (
    <ScrollView style={s.root} contentContainerStyle={s.content}>
      <Text style={s.icon}>☀️</Text>
      <Text style={s.city}>San Francisco</Text>
      <Text style={s.temp}>72°F</Text>
      <Text style={s.desc}>Sunny</Text>
      <View style={s.details}>
        {[['💧','45%','Humidity'],['💨','12 mph','Wind'],['👁','10 mi','Visibility']].map(([i,v,l])=>(
          <View key={l} style={s.detail}><Text style={s.dIcon}>{i}</Text><Text style={s.dVal}>{v}</Text><Text style={s.dLbl}>{l}</Text></View>
        ))}
      </View>
      <View style={s.forecast}>
        {days.map(d => (
          <View key={d.day} style={s.fDay}>
            <Text style={s.fLabel}>{d.day}</Text>
            <Text style={s.fIcon}>{d.icon}</Text>
            <Text style={s.fTemp}>{d.temp}°</Text>
          </View>
        ))}
      </View>
    </ScrollView>
  );
}

const s = StyleSheet.create({
  root: { flex: 1, backgroundColor: '#1565C0' },
  content: { alignItems: 'center', paddingVertical: 40 },
  icon: { fontSize: 80 },
  city: { fontSize: 28, color: '#fff', marginTop: 16 },
  temp: { fontSize: 64, color: '#fff' },
  desc: { fontSize: 18, color: '#90CAF9' },
  details: { flexDirection: 'row', marginTop: 40, width: '80%', justifyContent: 'space-around' },
  detail: { alignItems: 'center' },
  dIcon: { fontSize: 24 },
  dVal: { color: '#fff', fontWeight: '700', marginTop: 4 },
  dLbl: { color: '#90CAF9', fontSize: 12 },
  forecast: { flexDirection: 'row', marginTop: 40, width: '90%', justifyContent: 'space-evenly' },
  fDay: { alignItems: 'center' },
  fLabel: { color: '#90CAF9' },
  fIcon: { fontSize: 28, marginVertical: 4 },
  fTemp: { color: '#fff' },
});
""".trimIndent()
    ),
)
